04.ls/ls.rb

```diff
@@ -25,38 +25,34 @@ def main
 end
 
 def show_long_format(entries, path)
-  entries_long_format = []
+  long_format_entries = []
   total_blocks = 0
-
-  entries.each do |entry_name|
-    entry_details = File.lstat(entry_name)
-    total_blocks += entry_details.blocks
-    entries_long_format << load_details_each_entry(entry_name, entry_details, path)
+  entries.each do |entry|
+    lstat = File.lstat(entry)
+    total_blocks += lstat.blocks
+    long_format_entries << load_status_each_entry(entry, lstat, path)
   end
-  print_long_data(entries_long_format, total_blocks)
+  print_long_data(long_format_entries, total_blocks)
 end
 
-def load_details_each_entry(entry_name, entry_info, path)
-  xattr = Xattr::Lib.list(path + entry_name, @no_follow = true)
-  entry_details_array = []
-
-  entry_details_array << (!xattr.empty? ? "#{export_permission(entry_info)}@" : "#{export_permission(entry_info)} ")
-  entry_details_array << entry_info.nlink.to_s.rjust(3)
-  entry_details_array << Etc.getpwuid(entry_info.uid).name
-  entry_details_array << Etc.getgrgid(entry_info.gid).name.rjust(5)
-  entry_details_array << entry_info.size.to_s.rjust(5)
-  entry_details_array << format_time(entry_info.ctime)
-  entry_details_array <<
-    if entry_info.ftype == 'link'
-      "#{fetch_filetype(entry_name, path)} -> #{File.readlink(entry_name)}"
-    else
-      fetch_filetype(entry_name, path)
-    end
+def load_status_each_entry(entry, lstat, path)
+  xattr = Xattr::Lib.list(path + entry, @no_follow = true)
+  lstat_array = []
```

Comment:
```suggestion
  columns = []
```

とか。

ちなみに、`foo_array`と書くよりも、`foos`のように複数形にする方が正攻法な命名ですね。